src/cs/MonoGame.Extended/ViewportAdapters/ViewportAdapter.cs

```diff
@@ -4,10 +4,17 @@
 
 namespace MonoGame.Extended.ViewportAdapters
 {
+    using MonoGame.Extended.ViewportAdapters.Exceptions;
+
     public abstract class ViewportAdapter : IDisposable
     {
         protected ViewportAdapter(GraphicsDevice graphicsDevice)
         {
+            if (graphicsDevice is null)
+            {
+                throw new NullGraphicsDeviceException($"Attempted to construct {nameof(ViewportAdapter)} but {nameof(GraphicsDevice)} is null");
```

Comment:
This should just throw an `ArgumentNullException` since the issue here is that the *argument* is null, not that the graphics device itself is null.

Please use the new method for checking

```cs
ArgumentNullException.ThrowIfNull(graphicsDevice);
```